src/schema/v2/me/conversation/submit_inquiry_request_mutation.ts

```diff
@@ -0,0 +1,136 @@
+
+const InquiryItemType = new GraphQLUnionType({
+  name: "InquiryItemType",
+  types: [ArtworkType, ShowType],
+  resolveType: ({ __typename }) => {
+    switch (__typename) {
+      case "Artwork":
+        return ArtworkType
+      case "PartnerShow":
+        return ShowType
+      default:
+        return null
+    }
+  },
+})
+
+const InquiryQuestionInput = new GraphQLInputObjectType({
+  name: "InquiryQuestionInput",
+  fields: () => ({
+    questionID: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    details: {
+      type: GraphQLString,
+    },
+  }),
+})
+
+export const InquiryRequest = new GraphQLObjectType<any, ResolverContext>({
+  name: "InquiryRequest",
+  description: "A request to inquire on an artwork",
+  fields: () => ({
+    ...InternalIDFields,
+    inquireable: {
+      type: InquiryItemType,
+      resolve: (result) => {
+        const { inquireable_type, inquireable } = result
+        return {
+          __typename: inquireable_type,
+          ...inquireable,
+        }
+      },
+    },
+    inquirer: {
+      type: UserType,
+      resolve: ({ inquirer }, _, { userByIDLoader }) => {
+        const { id } = inquirer
+        return userByIDLoader(id)
+      },
+    },
+    shippingLocation: {
+      type: LocationType,
+      resolve: ({ inquiry_shipping_location }) => inquiry_shipping_location,
+    },
+    questions: {
+      type: new GraphQLList(GraphQLString),
+      resolve: ({ inquiry_questions }) => inquiry_questions,
+    },
+  }),
+})
+
+export const submitInquiryRequestMutation = mutationWithClientMutationId<
+  any,
+  any,
+  ResolverContext
+>({
+  name: "SubmitInquiryRequestMutation",
+  description: "Create an artwork inquiry request",
+  inputFields: {
+    inquireableID: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The inquireable object id (Artwork ID or Show ID)",
+    },
+    inquireableType: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The type of inquireable object (Artwork or Show)",
+    },
+    message: {
+      type: GraphQLString,
+      description: "Optional inquiry message",
+    },
+    questions: {
+      type: new GraphQLList(InquiryQuestionInput),
+      description: "List of structured inquiry questions",
+    },
+  },
+  outputFields: {
+    inquiryRequest: {
+      type: InquiryRequest,
+      description: "Artwork Inquiry request",
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: (
+    { inquireableID, inquireableType, message, questions },
+    { submitArtworkInquiryRequestLoader }
+  ) => {
+    if (inquireableType === "Artwork") {
+      if (!submitArtworkInquiryRequestLoader) return null
+      const locationQuestion = questions?.find(
+        (question) => question.questionID === "shipping_quote"
+      )
+      let inquiryShippingLocation
+      if (locationQuestion) {
+        try {
+          inquiryShippingLocation = JSON.parse(locationQuestion.details)
```

Comment:
Silly questions: this should support inquiry shipping location objects that look like this right?

```js
inquiry_shipping_location: {
    city: "Greenville",
    coordinates: [42.4153596, -74.0220769],
    country: "United States",
    postal_code: "12083",
    state: "New York",
    state_code: "NY"
}
```

Reply:
I believe so. The MP part just parses the stringified object coming from the client and passes it as a Hash to gravity. So as long as gravity accepts ☝️ format it should be fine. But if any data massaging is needed to be done in MP to convert client data structure to what gravity expects, we can follow up and add it here.